schema-registry/client/src/main/java/com/hortonworks/registries/schemaregistry/client/SchemaRegistryClient.java

```diff
@@ -1242,13 +1439,33 @@ public String run() {
                     return target.request(MediaType.APPLICATION_JSON_TYPE).get(String.class);
                 }
             });
-        } catch (LoginException e) {
+        } catch (LoginException | ProcessingException e) {
             throw new RegistryRetryableException(e);
         }
 
         return readEntity(response, clazz);
     }
 
+    private <T> T runWithFailoverContext(RetryBody<T> retryBody) {
+        WebTarget initialWebTarget = null;
+        RegistryRetryableException retryableException = null;
+        while (true) {
+            SchemaRegistryClient.SchemaRegistryTargets targets = currentSchemaRegistryTargets();
+            if (initialWebTarget == null) {
+                initialWebTarget = targets.rootTarget;
+            } else if (initialWebTarget.equals(targets.rootTarget)) {
+                throw retryableException;
```

Comment:
Why are you throwing exception here? Please explain it in a comment so that others can understand.

Reply:
Ok, will add more comments.

Reply:
This is the part of the code that prompted me in asking you to document how URLSelector and RetryPolicy work together.